src/gt4py/backend/base.py

```diff
@@ -77,6 +77,16 @@ class Backend(abc.ABC):
     #:  - "is_compatible_type": StorageLikeInstance -> bool
     storage_info: ClassVar[Optional[Dict[str, Any]]] = None
 
+    #: Language support:  Dict[str, Any]
+    #:
+    #:  - "primary": str Name of the primary language.
+    #:  - "secondaries": List[str] Names of supported secondary languages.
```

Comment:
I think these two concepts of "primary" and "secondary" language need a longer explanation, both in the user documentation and in the code docs. It is not really clear to me what "secondary" language means by just reading this description. Actually, even the names "primary" and "secondary" are not completely self-explanatory to me

Reply:
I was struggling with this a bit.

The backend generates source code in a certain language (Python / C++ / Cuda) which implements the stencil as optimally as you are going to get out of this backend. What do we call this language? I called it primary, because the primary target is the optimized stencil code.

Then, if the stencil is not generated in python directly, the backend generates bindings and a module in python, but conceptually might do the same for other languages if implemented and requested. That is the second language we come across in the generation / compilation process. I called it secondary here, because any code generated in this language will ultimately call the primary target, the optimized stencil.